packages/kbn-discover-utils/src/utils/build_data_record.ts

```diff
@@ -35,9 +35,13 @@ export function buildDataTableRecord(
  * @param docs Array of documents returned from Elasticsearch
  * @param dataView this current data view
  */
-export function buildDataTableRecordList(
+export function buildDataTableRecordList<T extends DataTableRecord = DataTableRecord>(
   docs: EsHitRecord[],
-  dataView?: DataView
+  dataView?: DataView,
+  { processRecord }: { processRecord?: (record: DataTableRecord) => T } = {}
```

Comment:
**nit**: as `dataView` previously was the last optional argument, this function signature made sense. But if we add a parameter and invoke it without having a data view, we then need to force passing a second argument to pass then a third one, and it'll start looking something like
```ts
buildDataTableRecordList(docs, undefined, {processRecord})
```
Shall we refactor the function to get named parameters as we are adding the processRecord now?
```ts
buildDataTableRecordList({docs, dataView, processRecord})
```

Reply:
I fully agree with this, but `buildDataTableRecordList` is used outside of Discover and I'd like to avoid pinging other teams for review on this PR. I can make this change in a followup though to avoid blocking this one.

Reply:
I opened a PR based on this one for the signature change here: https://github.com/elastic/kibana/pull/184976.